src/Entity/User.php

```diff
@@ -163,23 +158,40 @@ public function getAvailableAssignments(): array
 
         return $availableAssignments;
     }
-
+    
     /**
      * @throws AssignmentNotFoundException
      */
-    public function getAvailableAssignmentById(int $assignmentId): Assignment
+    public function getAssignmentById(int $assignmentId): Assignment
```

Comment:
IMO this logic should be moved to the assignments collection class. Something like findById or you could use the  matching(Criteria $criteria) from doctrine collections

Reply:
Yes it can be, but... Let's say as for me it would make sense in case we were asking for info from repository and returning it in a form of AssignmentCollection. But, it is a user entity, and method `getAssignmentById` means that we are asking for an assignment of the specific user by id. But let's also ask guys.
@Skyld , @luisfn , what do you think guys?

Reply:
In the action we should have only:

`$assignment = $user->getAssignmentById($assignmentId)` - throws `AssignmentNotFoundException` if assignment is not found, or `AssignmentUnavailableException` if it's not found

The methods in `User.php`:

```
/**
     * @return Assignment[]
     */
    public function getAvailableAssignments(): array
    {
        $availableAssignments = [];
        $now = Carbon::now()->toDateTime();

        foreach ($this->getAssignments() as $assignment) {
            if ($assignment->isCancelled()) {
                continue;
            }

            if (!$assignment->getLineItem()->isAvailableForDate($now)) {
                continue;
            }

            $availableAssignments[] = $assignment;
        }

        return $availableAssignments;
    }

    /**
     * @throws AssignmentNotFoundException
     * @throws AssignmentUnavailableException
     */
    public function getAvailableAssignmentById(int $assignmentId): Assignment
    {
        foreach ($this->getAvailableAssignments() as $assignment) {
            if ($assignment->getId() === $assignmentId) {

                if ($assignment->isAvailable()) {
                    return $assignment;
                }

                throw new AssignmentUnavailableException(
                    sprintf("Assignment with id '%s' for user '%s' is unavailable.", $assignmentId, $this->getUsername())
                );
            }
        }

        throw new AssignmentNotFoundException(
            sprintf("Assignment id '%s' not found for user '%s'.", $assignmentId, $this->getUsername())
        );
    }
```

Reply:
For me it is fine to have this method here.

`$user->getAssignmentById($assignmentId)`

is better than

`$user->getAssignments()->findById($id)`

Reply:
Ok Guys. Let's go with the @uncleempty solution then :)